src/lib/scaffolderAction/CustomScaffolderActionContext.ts

```diff
@@ -0,0 +1,55 @@
+import fetch from 'node-fetch';
+import { getLogger } from '@/logger';
+import { BaseLogger } from 'pino';
+import { ScaffolderActionContext } from '$/types';
+import { AGENT_SCAFFOLDER_LOG_PATH } from '@/scaffolderAction/constants';
+
+
+export class CustomScaffolderActionContext implements ScaffolderActionContext {
+  private readonly brokerClientUrl: string;
+  private readonly logger: BaseLogger;
+  private readonly actionId: string;
+
+
+  readonly workspacePath: string;
+  readonly payload: { body: Record<string, string>; localWorkspacePath: string };
+
+  constructor({ brokerClientUrl, actionId, payload }: { brokerClientUrl: string, actionId: string, payload: {
+      body: Record<string, string>, localWorkspacePath: string
+    }
+  }) {
+    this.brokerClientUrl = brokerClientUrl;
+    this.actionId = actionId;
+    this.payload = payload;
+    this.workspacePath = payload.localWorkspacePath;
+    this.logger = getLogger('RoadieAgentForwarder');
+  }
+
+
+  async log(content: string, context?: Record<string, string>) {
```

Comment:
Simple custom scaffolder context which exposes a logger, workspace and the payload passed into when the action was triggered